cmd/root/db.go

```diff
@@ -0,0 +1,135 @@
+package root
+
+import (
+	"database/sql"
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/NicksPatties/sweet/util"
+	_ "github.com/mattn/go-sqlite3"
+)
+
+// Gets a pointer to the stats database. If the database file
+// doesn't exist already, it will be created at sweet's default
+// configuration location (`~/.config/sweet`), or at the path
+// specified by `SWEET_DB_LOCATION`, if it's defined.
+//
+// If an error is returned from this function, then the pointer
+// will be `nil`.
+func SweetDb() (*sql.DB, error) {
+	// get the path for the database
+	var dbPath string
+	if envDir := os.Getenv("SWEET_DB_LOCATION"); envDir != "" {
+		dbPath = envDir
+	} else {
+		sweetDir, err := util.SweetConfigDir()
+		if err != nil {
+			return nil, fmt.Errorf("failed to find user config directory: %v", err)
+		}
+		dbPath = sweetDir
+	}
+
+	// create the sweet config directory
+	if err := os.MkdirAll(dbPath, 0775); err != nil {
+		return nil, fmt.Errorf("failed to find or create sweet config directory: %v", err)
+	}
+
+	// Open a connection to the SQLite database
+	db, err := sql.Open("sqlite3", path.Join(dbPath, "sweet.db"))
+	if err != nil {
+		db.Close()
+		return nil, fmt.Errorf("failed to open database: %v", err)
+	}
+
+	// Test the connection
+	err = db.Ping()
+	if err != nil {
+		db.Close()
+		return nil, fmt.Errorf("failed to connect to database: %v", err)
+	}
+
+	// create the reps table if it doesn't exist
+	_, err = db.Exec(`
+		CREATE TABLE if not exists reps(
```

Comment:
I need to verify that the values stored in this table **cannot be negative.** The following columns should be greater than or equal to 0.
- [ ] start
- [ ] end
- [ ] wpm
- [ ] raw
- [ ] dur
- [ ] acc
- [ ] miss
- [ ] errs